doc/manual.asciidoc

```diff
@@ -969,6 +971,31 @@ express the implicit dependency.)
 File paths are compared as is, which means that an absolute path and a
 relative path, pointing to the same file, are considered different by Ninja.
 
+[[validations]]
+Validations
+~~~~~~~~~~~
+Validations listed on the build line cause the specified files to be
+added to the top level of the build graph (as if they were specified
+on the Ninja command line) whenever the build line is a transitive
+dependency of one of the targets specified on the command line or a
+default target.
+
+Validations are added to the build graph regardless of whether the output
+files of the build statement are dirty are not, and the dirty state of
+the build statement that outputs the file being used as a validation
+has no effect on the dirty state of the build statement that requested it.
+
+A build edge can list another build edge as a validation even if the second
+edge depends on the first.
```

Comment:
Is this "circular" case covered in the test suite?

Also please add a test case with a cycle among the validation dependencies themselves.

Reply:
Yes, in BuildTest.ValidationDependsOnOutput.

I added BuildTest.ValidationCircular and BuildTest.ValidationWithCircularDependency, do those cover the cycles you were thinking of?

Reply:
Yes, thanks.